### PR TITLE
fix: merge .percy.yml config options into serializeDOM

### PR DESCRIPTION
**Pull request by rishigupta1599**
## Summary
- `.percy.yml` config options were not being passed to `PercyDOM.serialize()` — only per-snapshot options were used
- Now merges `utils.percy.config.snapshot` as defaults with per-snapshot options taking priority

## Test plan
- [ ] Verify existing tests pass
- [ ] Verify config-level settings (e.g. `enableJavaScript`) flow through to DOM serialization

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by rishigupta1599 — COMMENTED** [review]
Claude Code Review (automated) — 1 inline finding(s). Full report in the PR comment below. Verdict: Passed.

File: index.js [hunk 1]
```diff
@@ -62,6 +62,9 @@ module.exports = async function percySnapshot(t, name, options) {
     }
 
     // Serialize and capture the DOM
+    // Merge .percy.yml config options with snapshot options (snapshot options take priority).
```
Comment (rishigupta1599) [inline index.js:65]: **[Medium] Guard mergeSnapshotOptions for older sdk-utils**

`utils.mergeSnapshotOptions(forwardOpts)` is called unguarded, but the sibling utils calls just above (`isReadinessDisabled`, `getReadinessConfig`) use a `typeof utils.X === 'function'` fallback for older `@percy/sdk-utils`. On a version lacking this function, the call throws `TypeError`, which the outer try/catch swallows and logs as "Could not take DOM snapshot", silently dropping the snapshot.

**Suggestion:** Mirror the established pattern:
```suggestion
    const mergedOptions = typeof utils.mergeSnapshotOptions === 'function'
      ? utils.mergeSnapshotOptions(forwardOpts)
      : { ...(utils.percy?.config?.snapshot || {}), ...forwardOpts };
```

_Reviewer: stack:code-review_

**Comment by rishigupta1599** [comment]
# Claude Code PR Review

**PR:** https://github.com/percy/percy-testcafe/pull/670  •  **Head:** 6f86c56  •  **Reviewers:** stack:code-review

## Summary
Merges global `.percy.yml` snapshot config with per-snapshot options (per-call priority) via `utils.mergeSnapshotOptions(forwardOpts)` and passes the merged result as `serializeOpts` to `PercyDOM.serialize` before DOM capture, so config-level serialize options (e.g. `enableJavaScript`) reach the serializer. `readiness` remains stripped from the forwarded options.

## Review Table

| Priority | Category | Check | Status | Notes |
|----------|----------|-------|--------|-------|
| High | Security | No hardcoded secrets or credentials | N/A | Security lens skipped per orchestrator config. |
| High | Security | Authentication/authorization checks present | N/A | Security lens skipped per orchestrator config. |
| High | Security | Input validation and sanitization | N/A | Security lens skipped per orchestrator config. |
| High | Security | No IDOR — resource ownership validated | N/A | Security lens skipped per orchestrator config. |
| High | Security | No SQL injection (parameterized queries) | N/A | Security lens skipped per orchestrator config. |
| High | Correctness | Logic is correct, handles edge cases | Pass | Merge with per-call priority is correct; `readiness` stays stripped via `forwardOpts`. See F1 for an older-sdk-utils edge case. |
| High | Correctness | Error handling is explicit, no swallowed exceptions | Pass | Outer try/catch logs errors. Note: a missing `mergeSnapshotOptions` would be swallowed (F1). |
| High | Correctness | No race conditions or concurrency issues | Pass | No concurrency change; merge is synchronous before the existing `t.eval`. |
| Medium | Testing | New code has corresponding tests | Fail | No test asserts config options merge into `serializeOpts` (F2). |
| Medium | Testing | Error paths and edge cases tested | Fail | No coverage for config-present vs config-absent merge paths (F2). |
| Medium | Testing | Existing tests still pass (no regressions) | Pass | Change is additive to the serialize dependency; existing tests unaffected. |
| Medium | Performance | No N+1 queries or unbounded data fetching | Pass | Single shallow object spread; negligible. |
| Medium | Performance | Long-running tasks use background jobs | N/A | Not applicable to an SDK snapshot call. |
| Medium | Quality | Follows existing codebase patterns | Fail | `mergeSnapshotOptions` is called unguarded while sibling utils calls in the same function use a `typeof ... === 'function'` fallback (F1). |
| Medium | Quality | Changes are focused (single concern) | Pass | Tight, single-concern diff (3-line addition + one dependency swap). |
| Low | Quality | Meaningful names, no dead code | Pass | `mergedOptions` is clear; comments explain the why. |
| Low | Quality | Comments explain why, not what | Pass | Comment documents the `forwardOpts`/readiness rationale well. |
| Low | Quality | No unnecessary dependencies added | Pass | Reuses existing `@percy/sdk-utils` dependency. |

## Findings

- **File:** `index.js:65`
- **Severity:** Medium
- **Reviewer:** stack:code-review
- **Issue:** `utils.mergeSnapshotOptions(forwardOpts)` is invoked directly with no capability check. The two sibling calls just above it (`isReadinessDisabled`, `getReadinessConfig` at index.js:36-41) deliberately guard with `typeof utils.X === 'function' ? utils.X(...) : <local fallback>` to tolerate older `@percy/sdk-utils` versions. `mergeSnapshotOptions` was added in the same sdk-utils line; on an older installed version this call throws `TypeError: utils.mergeSnapshotOptions is not a function`, which the outer try/catch swallows and logs as "Could not take DOM snapshot", silently dropping the snapshot instead of falling back to `forwardOpts`.
- **Suggestion:** Mirror the established pattern: `const mergedOptions = typeof utils.mergeSnapshotOptions === 'function' ? utils.mergeSnapshotOptions(forwardOpts) : { ...(utils.percy?.config?.snapshot || {}), ...forwardOpts };`

- **File:** `tests/index.test.js`
- **Severity:** Low
- **Reviewer:** stack:code-review
- **Issue:** No test exercises the new merge behavior. The suite (94 lines) has no assertion that `.percy.yml` config snapshot options are merged into `serializeOpts`, nor that per-snapshot options win on key conflict, nor that the config-absent path still works.
- **Suggestion:** Add a test that sets `utils.percy.config.snapshot` (e.g. `{ enableJavaScript: true }`), calls `percySnapshot` with an overlapping option, and asserts the merged value reaching `PercyDOM.serialize` honors per-call priority. Cover the config-absent case too.

---

**Verdict: PASS** — Correct, focused change. Recommend adding the `typeof` guard for older sdk-utils parity (F1) and a merge test (F2); neither is blocking.

**Review by pranavz28 — APPROVED** [review]

**Review by Shivanshu-07 — COMMENTED** [review]
please add test cases
